### DISCUSSION TITLE
Add Hotend Only to Preheat Menu

### DISCUSSION THREAD

**Discussion by Cal-LF** [Jyers/Marlin#859] [General Discussion]
I find a "Hotend Only" checkbox would be useful when removing filament after print has finished and printer has cooled. The "Bed Only" checkbox is good for bed leveling. A hotend checkbox would be equally as good.

**Comment by nullstalgia** [comment]
Bonus points if it automatically unchecks the Bed Only when checked, and visa versa.

**Reply by Cal-LF** [reply]
Wow, that is just what is needed for the new preheat feature. Very nice.
Please add this as a pull request.

**Reply by embraceentropy** [reply]
Question about the blinking coordinates on the bottom of the display. Is the purpose of the blinking numbers to remind you to home the axes? If so, how did you add this feature?

**Reply by Cal-LF** [reply]
Last time I compiled Marlin bugfix 2.0, the coordinates are blinking.

**Reply by embraceentropy** [reply]
Interesting. If I'm compiling off of Jyers most recent changes should I see this behavior because I do not. Otherwise, what value/file should I focus on to add that back to my build?

**Reply by Cal-LF** [reply]
@embraceentropy Jyres FW does not blink the coordinates. I can't help with where to find this feature.

**Comment by nullstalgia** [comment]
tititopher68, you say replace with linked file but you sent over the entire
Marlin directory?

**Reply by nullstalgia** [reply]
Ah, I translated your previous message and I see now. 

Hope it works.

**Reply by Cal-LF** [reply]
Thanks for the encouragement. It is working nicely.

**Comment by Cal-LF** [comment]
Completed successful compile and flashed to printer. Works exactly like should.
I thank you so much. This feature is useful to me and surely for other users also.
Hope this is added into the next release.
tititopher68 - You sir are a genius along with Mr. Myers.

![Image01](https://user-images.githubusercontent.com/76499230/118875196-c0521f00-b8b9-11eb-8882-1f8b9bc17df0.jpg)
![Image02](https://user-images.githubusercontent.com/76499230/118875199-c0521f00-b8b9-11eb-89fc-531236dc1497.jpg)

**Comment by Jyers** [comment]
I have now implemented this with a slightly more elegant approach. Instead of having a bed only and hotend only option, you simply have the ability to enable or disable the heating of the bed and or the hotend. I believe this is a slightly cleaner approach while hopefully still meeting your needs.

@Cal-LF Let me know what you think

**Reply by Cal-LF** [reply]
@Jyers  Compiled successfully. Both the Blink Axis and the Preheat Bed/Hotend do the functions requested. I am able to use the preheat feature intuitively.
It looks like you have brought the blinking axis from Marlin bugfix 2.0. The appearance is similar with the ? marks and slower blinking.
I like the way @tititopher68  code works better but yours is a more broad implementation. 
Thank you for the time and effort given to this project.

**Reply by Jyers** [reply]
Just out of interests, what do you like better about his implementation? Always looking for ways to improve!

**Reply by Cal-LF** [reply]
@Jyers I was thinking a "hotend only" box would match what was already being used and the check boxes cleared the opposite box which is kind of slick. It will take a few uses to adjust to the new process. Having the boxes checked by default makes it easy to see what to do.
This update is really nice.

**Reply by Jyers** [reply]
One minute, I have a new concept I think you'll like :)

**Reply by Jyers** [reply]
@Cal-LF Give the newest version a try, its a lot cleaner!

**Reply by Cal-LF** [reply]
@Jyers Ok.

**Reply by Cal-LF** [reply]
@Jyers That menu fits right in with the rest of the adjustable settings. I like it more that the other layouts. 
A keeper in my opinion.

**Reply by nullstalgia** [reply]
+1 to this suggestion, it does make it *much* more consistent.

It might be a little fast for my personal preference, but I'll live with it ;)

Also, nice work on this implementation! Works very well.

**Reply by nullstalgia** [reply]
@tititopher68 I was agreeing with your suggestion :P

But I appreciate the better formatted instructions for others to use ❤️

**Reply by Jyers** [reply]
@tititopher68 Best not to assume before you know what you are dealing with. The reason my code is the way it is, is to better align with how marlin handles the blinking on other displays. I agree it needs a little tweaking to improve the fluidity, but I assure you this is the best approach. I'd advise you to be cautious of a "simpler is better" mindset, It's been the downfall of many a programmer.

**Reply by Cal-LF** [reply]
If @tititopher68 and @Jyers took the best of each feature, Blink Axis and Preheat, the JyresUI will improve. 
Mr. Myers has a lot to watch and make sure these changes will work with his vision of the final product.
Don't let egos get in the way of success. Sometimes language translation can be a barrier too.

**Reply by Jyers** [reply]
@Cal-LF Very well put!

**Reply by Jyers** [reply]
Looks like I should have held true to the marlin implementation. Turns out the one change I made to the blink function is what was causing the inconsistency, figures... haha. Anyway, it is now quite fluid and still blinks at a nice rate a little slower that tititopher's. Most importantly, it's independent from status refresh rate in case that ever needs to be adjusted. Thanks for the feedback everyone.
